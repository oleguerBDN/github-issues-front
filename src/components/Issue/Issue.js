import dompurify from "dompurify";
import styles from "./Issue.module.css";

const Issue = ({ title, body, state, author, authorUrl }) => (
  <div className={styles.issue}>
    <h1 className={styles["issue__title"]}>{title}</h1>
    <span
      className={
        state === "OPEN"
          ? styles["issue__state--open"]
          : styles["issue__state--closed"]
      }
    >
      {state}
    </span>
    <a className={styles["issue__author"]} href={authorUrl}>
      {author}
    </a>
    <div className={"markdown-body " + styles["issue__body"]}>
      <div dangerouslySetInnerHTML={{ __html: dompurify.sanitize(body) }} />
    </div>
  </div>
);

export default Issue;
