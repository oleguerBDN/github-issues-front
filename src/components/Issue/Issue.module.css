.issue {
  display: flex;
  flex-direction: column;
  color: #c9d1d9;
}

.issue__title {
  color: #161b22;
}

.issue__state--open {
  color: #090;
  font-weight: bolder;
}

.issue__state--closed {
  color: #b30000;
  font-weight: bolder;
}

.issue__author {
  margin-top: 10px;
  background-color: #161b22;
  padding: 10px;
  border: 1px solid #c9d1d9;
  border-radius: 10px 10px 0 0;
}

.issue__body {
  background-color: #0d1117;
  padding: 10px;
  border: 1px solid #c9d1d9;
}
