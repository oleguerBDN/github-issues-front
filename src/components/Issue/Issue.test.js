import { render, screen } from "@testing-library/react";
import Issue from "./Issue";
import ReactTestRenderer from "react-test-renderer";

describe("Given a Issue component", () => {
  describe("When it's called a title and author", () => {
    test("Then it should render both the given title and the author", () => {
      const expectedTitle = "useState bug";
      const expectedAuthor = "Karim";

      render(
        <Issue title={expectedTitle} author={expectedAuthor} authorUrl={""} />
      );
      const issueTitle = screen.getByRole("heading", { name: expectedTitle });
      const issueAuthor = screen.getByRole("link", { name: expectedAuthor });

      expect(issueTitle).toBeInTheDocument();
      expect(issueAuthor).toBeInTheDocument();
    });
  });

  describe("When it's called with an html body including a heading", () => {
    test("Then it should render this HTML", () => {
      const htmlBody = "<h4>Just a heading</h4>";
      const expectedTitle = "Just a heading";

      render(<Issue body={htmlBody} />);
      const htmlBodyHeading = screen.getByRole("heading", {
        name: expectedTitle,
      });

      expect(htmlBodyHeading).toBeInTheDocument();
    });
  });

  describe("When it's called with all the props", () => {
    test("Then it should render an Issue matching the snapshot", () => {
      const title = "Just a title for testing";
      const author = "Karim";
      const authorUrl = "https://www.karim.com";
      const state = "CLOSED";
      const htmlBody = "<p>Just an html body, nothing different</p>";

      const issue = ReactTestRenderer.create(
        <Issue
          title={title}
          author={author}
          authorUrl={authorUrl}
          state={state}
          body={htmlBody}
        />
      );

      expect(issue.toJSON()).toMatchSnapshot();
    });
  });
});
