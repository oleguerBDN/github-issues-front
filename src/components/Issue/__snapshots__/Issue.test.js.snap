// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Issue component When it's called with all the props Then it should render an Issue matching the snapshot 1`] = `
<div
  className="issue"
>
  <h1
    className="issue__title"
  >
    Just a title for testing
  </h1>
  <span
    className="issue__state--closed"
  >
    CLOSED
  </span>
  <a
    className="issue__author"
    href="https://www.karim.com"
  >
    Karim
  </a>
  <div
    className="markdown-body issue__body"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Just an html body, nothing different</p>",
        }
      }
    />
  </div>
</div>
`;
